Networking/Networking/ProductImageInBackground/MediaUploadSessionManager.swift

```diff
@@ -0,0 +1,164 @@
+            completion(.failure(error))
+        }
+    }
+
+    public func handleBackgroundSessionCompletion(_ completionHandler: @escaping () -> Void) {
+        backgroundCompletionHandler = completionHandler
+    }
+}
+
+extension MediaUploadSessionManager: URLSessionDataDelegate {
+    public func urlSession(_ session: URLSession,
+                           dataTask: URLSessionDataTask,
+                           didReceive data: Data) {
+        if let existingData = taskResponseData[dataTask.taskIdentifier] {
+            taskResponseData[dataTask.taskIdentifier] = existingData + data
+        } else {
+            taskResponseData[dataTask.taskIdentifier] = data
+        }
+    }
+
+    public func urlSession(_ session: URLSession,
+                           task: URLSessionTask,
+                           didCompleteWithError error: Error?) {
+        guard let uploadID = task.taskDescription else {
+            DDLogDebug("MediaUploadSessionManager- task completed without an upload identifier. Task identifier: \(task.taskIdentifier)")
+            return
+        }
+        defer {
+            taskResponseData.removeValue(forKey: task.taskIdentifier)
+        }
+
+        if let error = error {
+            DDLogError("⛔️ MediaUploadSessionManager- Upload failure for task (\(uploadID)): encountered error: \(error.localizedDescription)")
+            notifyCompletion(.failure(error), for: uploadID)
+            return
+        }
+
+        guard let httpResponse = task.response as? HTTPURLResponse else {
+            DDLogError("⛔️ MediaUploadSessionManager- Upload failure for task (\(uploadID)): " +
+                       "response is not a valid HTTPURLResponse. Actual response: " +
+                       "\(String(describing: task.response))")
+            notifyCompletion(.failure(BackgroundUploadError.invalidResponse), for: uploadID)
+            return
+        }
+
+        guard let data = taskResponseData[task.taskIdentifier] else {
+            DDLogError("⛔️ MediaUploadSessionManager- Upload failure for task (\(uploadID)): " +
+                       "missing response data for task with identifier \(task.taskIdentifier)")
+            notifyCompletion(.failure(BackgroundUploadError.invalidResponse), for: uploadID)
+            return
+        }
+
+        if !(200...299).contains(httpResponse.statusCode) {
+            DDLogError("⛔️ MediaUploadSessionManager- Upload failure for task (\(uploadID)): " +
+                       "unexpected HTTP status code \(httpResponse.statusCode). " +
+                       "Full response: \(httpResponse) Headers: \(httpResponse.allHeaderFields)")
+            notifyCompletion(.failure(BackgroundUploadError.invalidResponse), for: uploadID)
+            return
+        }
+
+        // Use MediaMapper to parse response
+        if let jsonString = String(data: data, encoding: .utf8) {
```

Comment:
I see that `jsonString` is unused. I believe we will use this JSON response in a future PR.